src/WebJobs.Extensions.DurableTask/ManagedIdentityTokenSource.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using Microsoft.Azure.Services.AppAuthentication;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Token Source implementation specific to Managed Identity Service.
+    /// </summary>
+    [DataContract]
```

Comment:
The requirement for using `[DataContract]` with the ITokenSource for serialization reasons is something we should consider documenting in the `IOrchestrationContext.CallHttpAsync` method comment (or at least on the `ITokenSource` interface documentation), so that users who are making their own token sources know about this requirement.